test/simple_mc_integration.cpp

```diff
@@ -0,0 +1,81 @@
+#include "doctest.h"
+#include "simple_MC_integration.hpp"
+#include "Eigen/Eigen"
+#include <vector>
+#include "cartesian_geom/cartesian_kernel.h"
+#include "hpolytope.h"
+#include "known_polytope_generators.h"
+#include "ode_solvers/oracle_functors.hpp"
+#include "random_walks/random_walks.hpp"
+#include <iostream>
+#include <fstream>
+#include "misc.h"
+
+typedef double NT;
+typedef Cartesian<NT> Kernel;
+typedef typename Kernel::Point Point;
+typedef std::vector<Point> Points;
+typedef HPolytope<Point> HPOLYTOPE;
+typedef VPolytope<Point> VPOLYTOPE;
+typedef boost::mt19937 RNGType;
+typedef BoostRandomNumberGenerator<RNGType, NT> RandomNumberGenerator;
+
+NT exp_N_Dim(Point X){
+	return exp(-X.squared_length()) ;
+}
+
+void call_test_simple_mc_over_hyperrectangle(){
+	srand(time(0));
+	std::cout << "\nTESTS FOR SIMPLE MC INTEGRATION OVER HYPER-RECTANGLES USING UNIFORM WALKS\n";
+
+	std::vector<NT> ll{-1,-1};
+	std::vector<NT> ul{1,1};
+	Point LL(2,ll), UL(2,ul);
+	simple_mc_integrate(exp_N_Dim,10000,LL,UL);
+
+	std::vector<NT> ll1{0,0};
+	std::vector<NT> ul1{2,2};
+	Point LL1(2,ll1), UL1(2,ul1);
+	simple_mc_integrate(exp_N_Dim,10000,LL1,UL1);
+
+
+}
+
+void call_test_simple_mc_over_polytope(){
```

Comment:
A general comment for the code structure of the PR: 

In the header files (hpp files) you should **not** return void. These are functions that are a part of a **library** and should be treated as such. 
The test methods (cpp files) can be void methods (and should be if they do not produce intermediate results). However, the `CHECK()` statements are still missing to compare between results / methods which deem the tests themselves of no use. The test code should contain these statements so that the CI system runs them and we make certain that the code does not have bugs each time.